pkg/stream/event_stream_test.go

```diff
@@ -0,0 +1,61 @@
+package stream
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Produce message", func() {
+	var (
+		kafkaTopic   = "test-topic"
+		writer       *KafkaWriter
+		ctrl         *gomock.Controller
+		mockProducer *MockKafkaProducer
+	)
+	BeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+		mockProducer = NewMockKafkaProducer(ctrl)
+		writer = &KafkaWriter{
+			kafkaProducer: mockProducer,
+			kafkaTopic:    kafkaTopic,
+		}
+	})
+
+	It("write message", func() {
+		key := []byte("my-key")
+		value := []byte("my-value")
+
+		expectedMessage := &kafka.Message{
+			TopicPartition: kafka.TopicPartition{
+				Topic:     &kafkaTopic,
+				Partition: kafka.PartitionAny,
+			},
+			Key:   key,
+			Value: value,
+		}
+		mockProducer.EXPECT().Produce(expectedMessage, nil).Times(1).Return(nil)
+
+		result := writer.Write(key, value)
+
+		Expect(result).To(BeNil())
```

Comment:
```suggestion
		err := writer.Write(key, value)

		Expect(err).To(BeNil())
```

Reply:
here and elsewhere. ``result`` is a very generic term that hides-away the true meaning of the returned value